### ISSUE TITLE
Accessibility of new history: labels

### ISSUE THREAD

**Issue by elichad** [galaxyproject/galaxy#13940]
**Describe the bug**

*Edit: oops, I thought I was testing on dev but I was actually on 22.01. Looks like a bunch of these have been fixed, yay*

- Form inputs (e.g. history/dataset name) should have [visible](https://axe.deque.com/rule-help/en/label-not-present) and [persistent](https://axe.deque.com/rule-help/en/label-is-placeholder) labels - i.e. not just placeholders which vanish when you type. 
- The 'tags' field in particular is easy to mistake for a non-interactive element and would benefit from a text label and the tag image being persistent outside the field.
- The accessible name for a dataset list item is `<hid>:<name> button View data button Edit attributes button...` - I am not sure if this is what's meant to happen when you have buttons inside a list like this.
- The '?' link within the expanded dataset isn't an accessible name for screen readers (most don't read punctuation by default so the label is effectively empty)

**Galaxy Version and/or server at which you observed the bug**
Galaxy Version: dev
Commit: 2f01b32b948a85e0ce5ef916aa7b66597ab94371

**Screenshots**
![Expanded dataset view](https://user-images.githubusercontent.com/20683271/169355240-b71d5601-c8c2-4c96-bcae-1d82fb3cfb86.png)

**Comment by elichad** [comment]
Took another look at dev, extra comment:
- advanced filter fields would be better aria-labelled by the text above them (e.g. "filter by name") rather than using just the placeholder text ("any name" - less helpful)

**Comment by nomadscientist** [comment]
Accessibility report notes:
- Keyboard navigation is laborious: Navigation could be improved with the addition of navigation shortcuts so that users can navigate more quickly between the three panels
- Not all of the functionality is accessible to keyboard and screen reader users e.g. the tooltips that describe the function of each of the tools along with renaming a history and copying datasets between histories.  
- A number of navigation icons did not have a text alternative, making them invisible to the screen reader.
- Much of the content of the central panel is displayed graphically, additional text descriptions might be necessary to ensure that students with a visual impairment can still achieve the learning outcomes.
- The central panel changes dynamically as a result of using the tools but screen reader users are not made aware that these changes have occurred.
- There is insufficient contrast between the text and the background e.g. dark green text on a light green background.
- The text and the entire page cannot be resized to 200% and remain readable.
- The progress of a data upload is indicated by a colour change alone, grey to yellow to green or red.

**Comment by nomadscientist** [comment]
Hi folks! I've lost track of this. Any updates on how the battle of accessibility is going?

**Comment by mvdbeek** [comment]
There's been some progress with https://github.com/galaxyproject/galaxy/pull/14481, but I'm sure there's also work remaining. You can check the current state on https://test.galaxyproject.org/

**Comment by nomadscientist** [comment]
I've had our people test stuff again on that tester version (as I think originally they were looking at an older version) and here's what they've come back with:

The website was tested against the [Web Content Accessibility Guidelines 2.1](https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fwww.w3.org%2FTR%2FWCAG21%2F&data=05%7C01%7Cwendi.bacon%40open.ac.uk%7Cffba7f90ed8e4a0bf20d08da9fceaa5e%7C0e2ed45596af4100bed3a8e5fd981685%7C0%7C0%7C637998007206054565%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=nt810izDd19YTNF1EFDpq%2FPsjbOdbVdLODSZAyN7otI%3D&reserved=0)
 
_Listed below are the instances where the Galaxy platform fails to meet the accessibility guidelines.  I’ve included examples for illustrative purposes but these are by no means exhaustive._
 
Success Criterion 1.1.1 Non-text Content
Not all non-text content has a text alternative that serves the equivalent purpose. E.g. icons are used as buttons but do not have alternative text. Data is presented graphically in the central panel without a detailed text alternative.
 
Success Criterion 1.3.1: Info and Relationships and Success Criterion 4.1.2 Name, Role, Value
Most user interface controls do not have a name that can be read by assistive technology. E.g. When changing the parameters of the ‘Filter by quality’ tool the ‘Quality cut-off’ edit box is visually labelled as such but this label cannot be programmatically determined and the screen reader simply reads ‘Edit’.  The ‘Run Workflow’ button is labelled most unhelpfully ‘Button’. 
 
Success Criterion 1.4.4 Resize text
text cannot be resized without assistive technology up to 200 percent without loss of content or functionality.
 
Success Criterion 2.1.1 Keyboard
Not all functionality is operable through a keyboard interface. E.g. Tooltips are not discoverable by the keyboard.  Metadata about a file cannot be obtained because a file name cannot be given keyboard focus.  A tool cannot be re-run because the ‘Run this job again’ icon cannot be given keyboard focus.  Datasets cannot be copied from one history to another because there is no keyboard alternative to the ‘click and drag’ operation.
 
Success Criterion 2.4.2 Page Titled
The webpage does not have a title that describes its topic or purpose.
 
Success Criterion 2.4.3 Focus Order
Focusable components do not receive focus in an order that preserves meaning and operability. E.g. When viewing the Fast QC report, selecting a link from the Summary, focus jumps to the bottom of the page instead of being retained on the selected link.  When the 'Upload data’ dialogue box is open, keyboard focus is not retained within the dialogue box. When the link to display the contents of a file or the results of an analysis is selected, the content is displayed in the central panel but the central panel does not receive focus.
 
Success Criterion 2.4.4 Link Purpose (In Context)
The purpose of each link cannot be determined from the link text alone. E.g. There are multiple weblinks with the text ‘Display’ and ‘Edit Attribute’.  Additional information is required so that users can determine what will be displayed or edited by selecting the link.
 
Success Criterion 2.4.7 Focus Visible
The keyboard focus indicator is not always clearly visible. E.g. In the ‘Upload data’ dialogue box the focus indicator is a slightly darker shade of grey against a paler grey button.  This is too subtle and needs to have better contrast.
 
Success Criterion 4.1.3 Status Messages
Status messages are not presented to the user by assistive technologies without receiving focus. E.g. The status of queued jobs is not read aloud by a screen reader.

**Comment by hexylena** [comment]
xred for 2.4.2, this one has bothered me for years.  https://github.com/galaxyproject/galaxy/issues/7914

**Comment by nomadscientist** [comment]
@dannon They used this link: https://test.galaxyproject.org/
Erm... recently...

I have asked for their methodology and will get back to you!

**Comment by nomadscientist** [comment]
Testing confirm that the target was [https://test.galaxyproject.org/](https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Ftest.galaxyproject.org%2F&data=05%7C01%7Cwendi.bacon%40open.ac.uk%7Ca74a299bff7145f4d87708daa13002eb%7C0e2ed45596af4100bed3a8e5fd981685%7C0%7C0%7C637999524815474233%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=F4XX5UPE7an7PbaPin1ZsVPsZtEsSWLKLzO%2BHM%2BpEao%3D&reserved=0). The testing was done manually by completing the tutorial at [https://training.galaxyproject.org/training-material/topics/introduction/tutorials/galaxy-intro-short/tutorial.html](https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Ftraining.galaxyproject.org%2Ftraining-material%2Ftopics%2Fintroduction%2Ftutorials%2Fgalaxy-intro-short%2Ftutorial.html&data=05%7C01%7Cwendi.bacon%40open.ac.uk%7Ca74a299bff7145f4d87708daa13002eb%7C0e2ed45596af4100bed3a8e5fd981685%7C0%7C0%7C637999524815474233%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=MLDmh3IRWZas3fXNH3bYincYPrBdUqEamOOSnLQ1kKM%3D&reserved=0)
 
Methodology was to check against each of the criteria listed below.

**Comment by mvdbeek** [comment]
As for the when, I suppose this was after I pointed to test.galaxyproject.org, which did run a recent dev including https://github.com/galaxyproject/galaxy/pull/14481
